### ISSUE TITLE
The Meta/Facebook SDK team has serious flow and management issues.

### ISSUE THREAD

**Issue by rmgomez** [facebook/facebook-sdk-for-unity#625]
I been using this SDK for a while, here are some points that need to be adressed.

1. The code is open source yet not a single PR has been merged from external developers. Facebook Developers should have some time to do code review. Ask for updates or close the PRs that can't be merged.
2. No one is closing or referencing the issues in their commits. This is very hard to track errors and the Issue tracker is just a mess, lots of duplicates, lots of outdated issues, no one responding on time.
3. Your QA process is just embarrassing, if this would be my team I would have fired lot's of people for releasing non-tested code in proper environment. Based on the issues we had and the other community members, I'm 100% sure you don't do any kind of integration testing or a simple smoke test before releasing.
4. Take a moment to calculate how much development time you are wasting for all the companies and studios that need this sdk integrated. It's really a lot.
5. Your SDK is not up to date with current standards. Period. ( check GameAnalytics who is doing a good job releasing with UPM and OpenUPM)

I hope that this "renewed support" (https://github.com/facebook/facebook-sdk-for-unity/issues/611#issuecomment-1148061969) can make some core changes.

@JordiFB @owaisfb @pengtfb @SergioGuerreroFB @JarWarren

**Comment by fahall** [comment]
I'm not onboard with the idea of firing anyone. This seems like a systemic problem, not the result of any individual.  That said, it does seem like this team is so resource constrained as to be unable to provide sufficient value to end users. 

### Unacceptable
- SDK not released quickly enough for users to stay compatible with Facebook API updates
- SDK frequently released in broken state (e.g. #623 & #619 ) 
- SDK sometimes released in state that crashes on init

### Problematic
- SDK not released automatically with latest changes (e.g. #619 appears to have had a fix for many days but is not yet released)
- SDK cannot be built with instructions provided (tested on Windows, MacOS, and Linux)
- Issues not marked resolved when code released to resolve them
- Issues frequently not responded to.
- SDK requires use of UnityPackage (deprecated install procedure)
- Not merging fixes from 3rd party devs
- No roadmap or ETA on critical bug fixes (let alone version updates or features).

Our team has easily spent tens if not hundreds of dev hours trying to do *basic* event tracking with this SDK. We just today got a build of our app using SDK 13.2.0 that compiles by writing our own Podfile generation code, but the app crashes upon FB.Init. So, despite all of this work, the SDK remains unusable. 

It would probably be a better PR move to stop officially supporting Unity than to offer the current level of SDK support.

**Comment by c0nd3v** [comment]
@JordiFB @JarWarren @SergioGuerreroFB @owaisfb @zpao @jawwad

**Comment by saxahan** [comment]
v14 for ios build output
`pod install output:

Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "FBSDKCoreKit":
  In Podfile:
    FBSDKCoreKit (~> 14.0)

None of your spec sources contain a spec satisfying the dependency: `FBSDKCoreKit (~> 14.0)`.`

The mentions above are absolutely right about THE META

**Comment by zenlita** [comment]
@saxahan I recommend that you try updating pod
https://stackoverflow.com/questions/67134352/cocoapods-could-not-find-compatible-versions-for-pod-fbsdkcorekit

**Comment by saxahan** [comment]
> @saxahan I recommend that you try updating pod https://stackoverflow.com/questions/67134352/cocoapods-could-not-find-compatible-versions-for-pod-fbsdkcorekit

No it doesn't work. Now it gives this error https://github.com/facebook/facebook-sdk-for-unity/issues/626

**Comment by zenlita** [comment]
@saxahan, updating pod resolved the pod install issue. I have commented on the issue #626. It is a separate situation where the configuration was not correct in Unity.

**Comment by zenlita** [comment]
I would also like to take the chance to request that further issues be reported here: https://developers.facebook.com/support/bugs. This is because this is our official channel of bug reporting and we are better set-up to triage problems. Appreciate the understanding and patience!

**Comment by fahall** [comment]
> I would also like to take the chance to request that further issues be reported here: https://developers.facebook.com/support/bugs. This is because this is our official channel of bug reporting and we are better set-up to triage problems. Appreciate the understanding and patience!

Historically, anything I have posted there regarding the Unity Facebook SDK has gone unread and unhandled. We stopped posting there and started posting here because it was the only place we could get a non-zero chance of response.

Are you planning to add resources to handle the requests submitted via ‘support/bugs’?

**Comment by zenlita** [comment]
@fahall I'm very sorry to hear that was your experience, this is not right. Please send me some examples of the issues that you posted that were gone unread and unhandled. I'll look into it.

**Comment by snorrsi** [comment]
@zenlita since you are working on improving your Unity support you can take a look at this case #584 
And also maybe close some of the other 300 something cases as they are most likely outdated.
And maybe create UPM compatible git package and tgz

**Comment by cometa93** [comment]
I totally agree with everything written here. It's not the first time Facebook SDK team breaks the projects. 
The responsibility on such a huge service, while most of industry is clearly dependent on the SDK made by you is really high. 

But the quality of service is the worst I've seen, you stand in the very same line as Appstore Connect Frontend devs.


Just take a look at the reaction time. You did release the project breaking package it's already 2 weeks after 14.0 release and no hotfix was released. The time wasted to fix what you should fix is unacceptable.

**Comment by MoonDawnGames** [comment]
Couldn't agree more with everything written here too. Probably one of the clumsiest team I have ever seen! None of you in the team realizes the importance of this system you are being tasked with to maintain. Furthermore, if you truly had a caring senior, he/she wouldn't allow any of you to be at this level of incompetence. Seriously, get your acts together!

**Comment by c0nd3v** [comment]
#ThisIsZuckedUp

**Comment by fahall** [comment]
@zenlita no response here: https://developers.facebook.com/community/threads/5313337695388980/

This one was eventually answered: https://developers.facebook.com/support/bugs/577269943610017/?join_id=f2f765c11de9a98

I want to take a moment to point out that I do not think that any one developer on this project is at fault for the unacceptable state of this SDK and its development. There are almost always larger contributing factors (e.g. process restrictions & resource allocation), and I suspect that leadership does not prioritize this SDK. 

I am not in support of blame or aggression shown by some of the other posters here. I understand their frustration, and I don't think frustration is an excuse for unkindness.

**Comment by zenlita** [comment]
Hi @fahall, thank you for the links. The first link is our community forum where we encourage the community of developers to help one another and is not officially tracked by any dedicated Meta team. The second is to our official bug support channel where a Meta team is dedicated to respond and triage the issues.

**Comment by derwaldgeist** [comment]
I have to second on what has being said. Trying to update my Unity 2019.4 installation, because Google now requires all apps to support Android 12 (API 31). It's a pure mess. Over 300 open issues, with nearly 0 response. If you are mainting your own bug tracker, why do you even let people open issues here on Github?! I have absolutely no idea which version of the plugin actually supports Android 12, as this is nowhere mentioned in the change log. Guys, this plugin is used in production! We're considering removing Facebook login altogether from our app now. Saves us a lot of trouble and dev time.

**Comment by derwaldgeist** [comment]
> Still cannot get **.xcworkspace** 😞

Were you ever able to resolve this? I finally managed to get this SDK working on Android with 15.0.0 - only to learn it broke my iOS builds because of that stupid FBSDKCoreKit (~> 15.1) thing.

I upgraded my cocoapods installation and even added UTF8 encoding to my bash shell. But still, this error occurs.

@ maintainers: The QA of this plugin is just horrible.

**Comment by derwaldgeist** [comment]
Ok. I found the root cause for the FBSDKCoreKit (~> 15.1) error. It would only work if the deployment target of the project was set to iOS 12. Mine was on iOS 11. Why hasn't this been mentioned in the change logs?! In the docs, the minimum iOS version is even 8...

**Comment by Andreyul** [comment]
> ... Why hasn't this been mentioned in the change logs?! In the docs, the minimum iOS version is even 8...

https://github.com/facebook/facebook-ios-sdk/blob/main/CHANGELOG.md#1500
> The minimum supported version of iOS and tvOS is now 12.0.

**Comment by derwaldgeist** [comment]
> > ... Why hasn't this been mentioned in the change logs?! In the docs, the minimum iOS version is even 8...
> 
> https://github.com/facebook/facebook-ios-sdk/blob/main/CHANGELOG.md#1500
> 
> > The minimum supported version of iOS and tvOS is now 12.0.

This is the Facebook iOS SDK. I mean the change log of the Unity SDK.

BTW: I just learned that the latest version of this SDK builds for iOS, but then crashes immediately on the device, although I am using a recommended Unity version (2019.4.40). This plugin is one of the most flawed pieces of software I ever encountered. So frustrating. Unbelievable.

**Comment by kdeger** [comment]
We just received a mail which informs us ("Your app will be affected by the deprecation of Marketing API versions prior to v15.0.") that Facebook will be deprecating all versions prior to v15.0 on April 25th. Just tried to upgrade the SDK to 16.0 and yet the issue remains the same. How can we upgrade the SDK version if none have been working for iOS since v15.0? Or should we wait for a new version which fixes the relevant issue before April 25th?

**Comment by derwaldgeist** [comment]
These guys are insane. The SDK is buggy as f..., and they even have the guts to force us to an update. I'm glad that I finally managed to upgrade this to 15.1, but I definitely do not want to upgrade it anytime soon.

Let alone that their whole analytics stuff sucks compared to Google Analytics. So we don't use it anymore and only need this SDK for Facebook Login. Still, we have to include this beast with all of that other stuff that does not give us any business value. And only for the "elderly people" who are still on Facebook.

**Comment by Thaina** [comment]
Facebook always demand us to response to them and update our app for them. They never do this on their own with the package they provide us
